src/Crypto/AuthenticationSecret.php

```diff
@@ -0,0 +1,63 @@
+<?php declare(strict_types = 1);
+
+namespace Cmnty\Push\Crypto;
+
+use InvalidArgumentException;
+
+class AuthenticationSecret implements RawBytes
+{
+    /**
+     * @var BinaryString
+     */
+    private $binaryString;
+
+    /**
+     * Create authentication secret.
+     *
+     * @param BinaryString $binaryString
+     *
+     * @throws InvalidArgumentException When the authentication secret is not the correct length.
+     */
+    public function __construct(BinaryString $binaryString)
+    {
+        if ($binaryString->getLength() != 16) {
+            throw new InvalidArgumentException('AuthenticationSecret could not be created: incorrect length.');
+        }
+
+        $this->binaryString = $binaryString;
+    }
+
+    /**
+     * Create authentication secret.
+     *
+     * @param string $base64UrlEncoded
+     *
+     * @return self
+     *
+     * @throws InvalidArgumentException When the authentication secret is not the correct length.
+     */
+    public static function createFromBase64UrlEncodedString(string $base64UrlEncoded) : self
+    {
+        return new self(BinaryString::createFromBase64UrlEncodedString($base64UrlEncoded));
+    }
+
+    /**
+     * Get raw bytes.
+     *
+     * @return string
+     */
+    public function getRawBytes() : string
+    {
+        return $this->binaryString->getRawBytes();
+    }
+
+    /**
+     * Get base 64 url safe sting.
+     *
+     * @return string
+     */
+    public function getBase64UrlEncodedString() : string
```

Comment:
Minor; "getBase64EncodedUrl" is perhaps a better method name. (it's in more files).
Or you have a good reason to add "string" to the name?